Kernel/platform/platform-rpipico/devices.c

```diff
@@ -52,21 +66,20 @@ static void timer_tick_cb(unsigned alarm)
         update_us_since_boot(&next, time_us_64() + (1000000 / TICKSPERSEC));
         hardware_alarm_set_target(0, next);
     }
-
+    irqflags_t irq = di();
+    udata.u_ininterrupt = 1;
+    tty_interrupt();
     timer_interrupt();
-
-    if (usbconsole_is_readable())
-    {
-        uint8_t c = usbconsole_getc_blocking();
-        tty_inproc(minor(BOOT_TTY), c);
-    }
+    udata.u_ininterrupt = 0;
+	irqrestore(irq);
 }
 
 void device_init(void)
 {
+    kprintf("Unallocated RAM: %dkB\n", UNUSEDSIZE / 1024);
+
     /* The flash device is too small to be useful, and a corrupt flash will
```

Comment:
Hm... I don't think Github clearly shows which line you're reviewing.
If you're talking about the comment, I didn't leave it..

As for Unallocated RAM, it's specifically for RAM, not flash. I tried to remove static allocation of userspace memory in swapper.c (e.g. `char progbase[USERMEM];`) but had issues with Pico hard faulting, and I'm new to arm to track it down, and I'm leaving it for later. So instead I just left myself a reminder for the case where USERMEM can be increased because there's unallocated space before stack bank.